src/shared/container-engine-lib.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# container engine helper functions to handle both podman and docker commands
+
+CICD_TOOLS_COMMON_LOADED=${CICD_TOOLS_COMMON_LOADED:-1}
+
+if [ "$CICD_TOOLS_CONTAINER_ENGINE_LOADED" -eq 0 ]; then
+    return 0
+fi
+
+if [ "$CICD_TOOLS_COMMON_LOADED" -ne 0 ]; then
+    source "${CICD_TOOLS_WORKDIR}/src/shared/common.sh"
+fi
+
+echo "loading container engine"
+
+CONTAINER_ENGINE_CMD=''
+
+container_engine_cmd() {
+
+    if [ -z "$(get_container_engine_cmd)" ]; then
+        if ! set_container_engine_cmd; then
+            return 1
+        fi
+    fi
+
+    if [ "$(get_container_engine_cmd)" = "podman" ]; then
+        podman "$@"
+    else
+        docker "--config=${DOCKER_CONF}" "$@"
+    fi
+}
+
+get_container_engine_cmd() {
```

Comment:
If worried about stripping newlines, what if you just re-define `CONTAINER_ENGINE_CMD` at the top and just use the variable instead of needing this func?

`CONTANER_ENGINE_CMD=$(echo -n $CONTAINER_ENGINE_CMD)`

Reply:
yup - actually getting rid of this function entirely, seems utterly redundant.